content/sdk/10.js/00.ethers/20.v6/01.features.md

```diff
@@ -121,3 +111,54 @@ const sentTx = await sender.sendTransaction({
   },
 });
 ```
+
+### Explanation
+
+1. **Set Greeting**: Define the new greeting message.
+
+    ```javascript
+    const greeting = "a new greeting";
+    ```
+
+2. **Populate Transaction**: Prepare the transaction for setting the greeting.
+
+    ```javascript
+    const tx = await greeter.populateTransaction.setGreeting(greeting);
+    ```
+
+3. **Calculate Gas**:
+    - Retrieve the current gas price.
+    - Estimate the gas limit for the transaction.
+    - Calculate the total fee.
+
+    ```javascript
+    const gasPrice = await sender.provider.getGasPrice();
+    const gasLimit = await greeter.estimateGas.setGreeting(greeting);
+    const fee = gasPrice.mul(gasLimit);
+    ```
+
+4. **Generate Paymaster Parameters**: Set up the paymaster parameters required for zkSync.
+
+    ```javascript
+    const paymasterParams = utils.getPaymasterParams(testnetPaymaster, {
+      type: "ApprovalBased",
+      token,
+      minimalAllowance: fee,
+      innerInput: new Uint8Array(),
+    });
+    ```
+
+5. **Send Transaction**: Include zkSync-specific fields in the `customData` object.
+
+    ```javascript
+    const sentTx = await sender.sendTransaction({
+      ...tx,
+      maxFeePerGas: gasPrice,
+      maxPriorityFeePerGas: 0n,
+      gasLimit,
+      customData: {
+        gasPerPubdata: utils.DEFAULT_GAS_PER_PUBDATA_LIMIT,
+        paymasterParams,
+      },
+    });
+    ```
```

Comment:
This is duplicating code which makes it harder to maintain. Can we move the explanations as comments inside the original code above?